### PR TITLE
NOISSUE: a little more comments for Flow

### PR DESCRIPTION
**Pull request by afiskon**

### CODE REVIEW

File: insolar/flow/flow.go [hunk 1]
```diff
@@ -23,14 +23,19 @@ import (
...
+// After creating a Handle you can register it in MessageBus like this:
+// `h.Bus.MustRegister(insolar.TypeGetObject, createdHandle.WrapBusHandle)`
+// You can find an example in insolar/ladger/artifactmanager/handler.go
 type Handle func(context.Context, Flow) error
 
 // MakeHandle is a function that constructs new Handle.
 type MakeHandle func(bus.Message) Handle
 
 // Procedure is a task that can execute itself.
+// Procedure is supposed to be executed in one pulse. If you have multiple steps that can be executed
```
Comment (andreyromancev) [inline insolar/flow/flow.go]: "supposed to be executed without pulse interruptions". Currently, sounds like it **must** complete within single pulse.

File: insolar/flow/flow.go [hunk 2]
```diff
@@ -46,9 +51,12 @@ type Flow interface {
 	// If cancellation happens during Handle execution, ErrCancelled will be returned.
 	Handle(context.Context, Handle) error
 
-	// Procedure starts routine and blocks Handle execution until cancellation happens or routine returns.
-	// If cancellation happens first, ErrCancelled will be returned.
+	// Procedure starts a routine and blocks Handle execution until cancellation happens or routine returns.
+	// If cancellation happens first, ErrCancelled will immediately be returned to the Handle. The Procedure
+	// continues to execute in the background, but it's return value will be discarded.
```
Comment (andreyromancev) [inline insolar/flow/flow.go]: "it's return value must be discarded"

File: insolar/flow/flow.go (hunk 1 above)
Comment (afiskon) [reply insolar/flow/flow.go]: OK, the wording changed.

File: insolar/flow/flow.go (hunk 2 above)
Comment (afiskon) [reply insolar/flow/flow.go]: OK, the wording changed.

File: insolar/flow/flow.go [hunk 3]
```diff
@@ -23,14 +23,20 @@ import (
...
+// After creating a Handle you can register it in MessageBus like this:
+// `h.Bus.MustRegister(insolar.TypeGetObject, createdHandle.WrapBusHandle)`
+// You can find an example in insolar/ladger/artifactmanager/handler.go
 type Handle func(context.Context, Flow) error
 
 // MakeHandle is a function that constructs new Handle.
 type MakeHandle func(bus.Message) Handle
 
 // Procedure is a task that can execute itself.
+// Please not that the Procedure is canceled if a pulse happens during it's execution.
```
Comment (andreyromancev) [inline insolar/flow/flow.go]: Please not**e** that Procedures are **not** terminated on pulse change.

File: insolar/flow/flow.go [hunk 4]
```diff
@@ -23,14 +23,20 @@ import (
...
+// Thus if you have multiple steps that can be executed in different pulses split them into separate Procedures.
+// Otherwise join the steps to a single Procedure.
```
Comment (andreyromancev) [inline insolar/flow/flow.go]: Otherwise join the steps **in**to a single Procedure.

File: insolar/flow/flow.go [hunk 5]
```diff
@@ -46,9 +52,12 @@ type Flow interface {
 	// If cancellation happens during Handle execution, ErrCancelled will be returned.
 	Handle(context.Context, Handle) error
 
-	// Procedure starts routine and blocks Handle execution until cancellation happens or routine returns.
-	// If cancellation happens first, ErrCancelled will be returned.
+	// Procedure starts a routine and blocks Handle execution until cancellation happens or routine returns.
+	// If cancellation happens first, ErrCancelled will immediately be returned to the Handle. The Procedure
+	// continues to execute in the background, but it's return value must be discarded by the Flow implementation.
```
Comment (andreyromancev) [inline insolar/flow/flow.go]: must be discarded by the Handle.

File: insolar/flow/flow.go (hunk 3 above)
Comment (afiskon) [reply insolar/flow/flow.go]: ok

File: insolar/flow/flow.go (hunk 4 above)
Comment (afiskon) [reply insolar/flow/flow.go]: ok

File: insolar/flow/flow.go (hunk 5 above)
Comment (afiskon) [reply insolar/flow/flow.go]: ok.

**Pull request merged**